plugins/index.js

```diff
@@ -90,8 +91,16 @@ export function loadPlugin(manifest) {
         script.src = getSiteURL() + bundlePath;
         script.onload = onLoad;
         console.log('Loading ' + manifest.id + ' plugin'); //eslint-disable-line no-console
-        document.getElementsByTagName('head')[0].appendChild(script);
 
+        if (!tryRequire(bundlePath)) {
```

Comment:
Can you walk me through how this works? We haven't added the script to the page yet (i.e. it hasn't executed), so I'm curious how this `require` "passes" as such.

Reply:
This `require` is actually loading the `bundlePath` javascript file at the time is executed. If it cannot find it then it fails and we do not continue to add the script to the webpage. 
If the plugin libraries were know before hand, this would be the regular `import` or `require` done before adding a module or library. Given that they are loaded dynamically later on, you can think of this `require` as a deferred import check.

The other approach if `require` feels overkill, I can handle the general [`onerror` event](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror) when it tries to load the script and log the error in the event handler.

Reply:
I've removed the `tryRequire` check given that not only was a duplicate load of the script but was causing unexpected behavior in development environment. I've included the `onerror` event handling that would gracefully handle any loading/executing errors for the script.